lib/web_ui/lib/src/engine/util.dart

```diff
@@ -166,7 +172,14 @@ TransformKind transformKindOf(List<double> matrix) {
   if (isIdentityTransform) {
     return TransformKind.identity;
   } else {
-    return TransformKind.transform2d;
+    // Check for scale, skew, and rotation.
+    final bool isTranslation2d =
```

Comment:
We should also probably check that `m[3]` and `m[7]` are zero here too, right (that is, the last element of the x and y column)? Otherwise this could do more than just a trivial translation.

Reply:
True!

Reply:
Oh wait, I think it's captured by the `isSimple2dTransform` check above, no?

Reply:
Oh you're right.